### PR TITLE
chore(v3): migrate graph-connector-bot to V3

### PR DESCRIPTION
**Pull request by formulahendry**

### CODE REVIEW

File: graph-connector-bot/infra/azure.bicep [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    }
+  }
+}
+
+resource webAppSettings 'Microsoft.Web/sites/config@2021-02-01' = {
+  name: '${webAppName}/appsettings'
+  properties: {
+    WEBSITE_NODE_DEFAULT_VERSION: '~16'
+    SCM_SCRIPT_GENERATOR_ARGS: '--node'
+    BOT_ID: botAadAppClientId
```
Comment (blackchoey) [inline graph-connector-bot/infra/azure.bicep:71]: maybe we can set these settings in line 47 directly?

File: graph-connector-bot/infra/azure.bicep [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  name: 'Azure-Bot-registration'
+  params: {
+    resourceBaseName: resourceBaseName
+    botAadAppClientId: botAadAppClientId
+    botAppDomain: webApp.properties.defaultHostName
+    botDisplayName: botDisplayName
+  }
+}
+
+// The output will be persisted in .env.{envName}. Visit https://aka.ms/teamsfx-provision-arm#output for more details.
```
Comment (blackchoey) [inline graph-connector-bot/infra/azure.bicep:93]: the aka link is incorrect, use teamsfx-actions/arm-deploy instead

File: graph-connector-bot/infra/azure.parameters.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "resourceBaseName": {
+      "value": "bot${{RESOURCE_SUFFIX}}"
```
Comment (blackchoey) [inline graph-connector-bot/infra/azure.parameters.json:6]: suggest give it a better name

File: graph-connector-bot/teamsapp.yml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    with:
+      appPackagePath: ./build/appPackage/appPackage.${{TEAMSFX_ENV}}.zip # Relative path to this file. This is the path for built zip file.
+    # Output: following environment variable will be persisted in current environment's .env file.
+    # TEAMS_APP_ID: the id of Teams app
+  - uses: teamsApp/publishAppPackage # Publish the app to Teams Admin Center (https://admin.teams.microsoft.com/policies/manage-apps) for review and approval
+    with:
+      appPackagePath: ./build/appPackage/appPackage.${{TEAMSFX_ENV}}.zip
+  # Output: following environment variable will be persisted in current environment's .env file.
+  # TEAMS_APP_PUBLISHED_APP_ID: app id in Teams tenant app catalog.
+projectId: e40c350a-84fb-4647-accd-3bd9840db224
```
Comment (blackchoey) [inline graph-connector-bot/teamsapp.yml:96]: should remove projectId

File: graph-connector-bot/infra/azure.bicep (hunk 1 above)
Comment (formulahendry) [reply graph-connector-bot/infra/azure.bicep:71]: As discuss offline, could not move `webApp.properties.defaultHostName`. So just keep it with no change,

**Review by blackchoey — APPROVED** [review]

**Pull request merged**